### PR TITLE
[COASTAL-97] (WIP) Alert at time of day

### PR DESCRIPTION
**Pull request by nhamming**
https://tidepool.atlassian.net/browse/COASTAL-97

@rickpasetto since you built the alerting system, I wanted to get your input on this addition. It allows for an alert at a specific time of day and it may repeat. 

Also see: https://github.com/tidepool-org/Loop/pull/468

### CODE REVIEW

File: LoopKit/Alert.swift [hunk 1]
```diff
@@ -32,6 +32,10 @@ public struct Alert: Equatable {
         case delayed(interval: TimeInterval)
         /// Delay triggering the alert by `repeatInterval`, and repeat at that interval until cancelled or unscheduled.
         case repeating(repeatInterval: TimeInterval)
+        /// Delay triggering until specific time of day, and no repeat
```
Comment (rickpasetto) [inline LoopKit/Alert.swift:35]: [nit] My first reaction is, why not just have `case daily(at: DateComponents, repeating: Bool)`? But that's minor.

File: LoopKit/Alert.swift [hunk 2]
```diff
@@ -32,6 +32,10 @@ public struct Alert: Equatable {
...
+        case dailyOnce(at: DateComponents)
```
Comment (rickpasetto) [inline LoopKit/Alert.swift:36]: May want to mention that only `hours`, `minutes`, and `seconds` are supported in the given `DateComponents`

**Review by rickpasetto — COMMENTED** [review]
Left some comments... some requiring consideration, but none really blocking

**Comment by nhamming** [comment]
@rickpasetto is adding this feature with https://github.com/tidepool-org/LoopKit/pull/440

**Pull request closed without merging**